tests/rest/client/v2_alpha/test_register.py

```diff
@@ -16,10 +17,13 @@ def make_homeserver(self, reactor, clock):
         self.url = b"/_matrix/client/r0/register"
 
         self.hs = self.setup_test_homeserver()
+        self.hs.clock = MockClock()
```

Comment:
instead of mockclock, you want `self.reactor.advance(time)`

Reply:
Oooh I didn't know that was a thing. Thanks!

Reply:
Thinking back about it, all of the calculation involved when rate limiting is done based on the HS clock, not the reactor's clock, so not sure it's the right solution.

Reply:
> all of the calculation involved when rate limiting is done based on the HS clock, not the reactor's clock

this is a bad thing, but the hs clock should be a thin shim around the reactor clock anyway?

Reply:
Right, I would not see it working because of a line I forgot to remove that was overwriting the clock. Fixed in ` [e11a462](/matrix-org/synapse/pull/4735/commits/e11a4620e2dd02380c94d4dae73a6eff8d82aa40)`